sui/src/config/mod.rs

```diff
@@ -105,6 +107,31 @@ fn socket_addr_from_hostport(host: &str, port: u16) -> SocketAddr {
         .unwrap_or_else(|| panic!("Hostname/IP resolution failed for {host}"))
 }
 
+// When validators start they need to find peers in the committee
+// Here we give them some time to retry connecting with exponentially back-off
+// TODO: for testnet, we may need to tune the parameters to allow
+// more asynchronous starting.
+fn socket_addr_from_hostport_retry(host: &str, port: u16) -> SocketAddr {
```

Comment:
I'm kind of worried this contains many magic numbers. In Narwhal, we set up a [`RetryConfiguration`](https://github.com/MystenLabs/narwhal/blob/main/network/src/retry.rs) object (with a sensible default). If Sui was to reuse it, it could pass those values in config and have them applied everywhere.
/cc @asonnino

Reply:
This functionality will be unneeded after https://github.com/MystenLabs/sui/pull/1890 lands

Reply:
to elaborate, after than change lands there is no hostname resolution that happens at config time. instead its deferred until a transaction actually needs to be issued to consensus.